torchvision/utils.py

```diff
@@ -382,6 +389,132 @@ def draw_keypoints(
     return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1).to(dtype=torch.uint8)
 
 
+# Flow visualization code adapted from https://github.com/tomrunia/OpticalFlow_Visualization
+@torch.no_grad()
+def flow_to_image(
+    flow_uv: torch.Tensor,
+    clip_flow: Optional[float] = None,
+) -> torch.Tensor:
+
+    """
+    Converts the two dimensional flow of image to a RGB Image.
+
+    Args:
+        flow_uv (Tensor): Flow UV image of shape [2, H, W]
+        clip_flow (float, optional): Clip maximum of flow values. Defaults to None.
+
+    Returns:
+        img (Tensor[3, H, W]): Image Tensor with flow drawn.
+    """
+
+    if flow_uv.ndim != 3:
+        raise ValueError("Input flow should have 3 dimensions [2, H, W]")
+    if flow_uv.size(0) != 2:
+        raise ValueError("Input flow should have shape [2, H, W]")
+
+    if clip_flow is not None:
+        flow_uv = torch.clip(flow_uv, 0, clip_flow)
+
+    u = flow_uv[0, :, :]
+    v = flow_uv[1, :, :]
+    rad = torch.sqrt(torch.square(u) + torch.square(v))
+    rad_max = torch.max(rad)
+    epslion = 1e-5
+    u = u / (rad_max + epslion)
+    v = v / (rad_max + epslion)
+    return _flow_uv_to_colors(u, v)
+
+
+@torch.no_grad()
+def _flow_uv_to_colors(
+    u: torch.Tensor,
+    v: torch.Tensor,
+) -> torch.Tensor:
+
+    """
+    Applies the flow color wheel to (possibly clipped) flow components u and v.
+
+    Args:
+        u (torch.Tensor): Input horizontal flow of shape [H, W]
+        v (torch.Tensor): Input vertical flow of shape [H, W]
+
+    Returns:
+       img (Tensor[3, H, W]): Flow visualization image.
+    """
+
+    flow_image = torch.zeros((3, u.shape[0], u.shape[1]), dtype=torch.uint8)
+    colorwheel = _make_colorwheel()
+    # shape [55x3]
+    ncols = colorwheel.shape[0]
+    rad = torch.sqrt(torch.square(u) + torch.square(v))
+
+    a = torch.atan2(-v, -u) / torch.pi
+    fk = (a + 1) / 2 * (ncols - 1)
+    k0 = torch.floor(fk).to(torch.long)
+    k1 = k0 + 1
+    k1[k1 == ncols] = 0
+    f = fk - k0
+
+    for i in range(colorwheel.shape[1]):
+        tmp = colorwheel[:, i]
+        col0 = tmp[k0] / 255.0
+        col1 = tmp[k1] / 255.0
+        col = (1 - f) * col0 + f * col1
+        idx = rad <= 1
+        col[idx] = 1 - rad[idx] * (1 - col[idx])
+        col[~idx] = col[~idx] * 0.75  # out of range
+        ch_idx = i
+        flow_image[ch_idx, :, :] = torch.floor(255 * col)
+    return flow_image
+
+
+def _make_colorwheel() -> torch.Tensor:
+    """
+    Generates a color wheel for optical flow visualization as presented in:
+    Baker et al. "A Database and Evaluation Methodology for Optical Flow" (ICCV, 2007)
+    URL: http://vision.middlebury.edu/flow/flowEval-iccv07.pdf.
+
+    Returns:
+        colorwheel (Tensor[255, 3]): Colorwheel Tensor.
```

Comment:
are you sure that this is 255? According to the code below this should be 55

Reply:
It's 55 my bad